dialogue-apache-hc5-client/src/main/java/com/palantir/dialogue/hc5/InstrumentedDnsResolver.java

```diff
@@ -32,15 +37,58 @@
 
     @Override
     public InetAddress[] resolve(String host) throws UnknownHostException {
+        boolean debugLoggingEnabled = log.isDebugEnabled();
+        long startNanos = debugLoggingEnabled ? System.nanoTime() : -1L;
```

Comment:
Is it worth noting that we're explicitly not using timers here as the overhead is undesirable for the common case?

Reply:
Good idea